### ISSUE TITLE
Validate configuration before parsing commands

### ISSUE THREAD

**Issue by dimitrystd** [Smartling/smartling-cli#49]
_Feedback from user_
**Steps**
Run `smartling-cli` from non configured folder (the current one or parents don't have config) use **invalid** command
```
$ smartling-cli projects
Usage:
  smartling [options] [-v]... init --help
  smartling [options]....
```

**Actual**:
You see help and you think that everything is fine with config and you just need to fix command

**Expected**:
Show error about config because if i will fix command then only after this i will get error about config

**Comment by dimitrystd** [comment]
At some point `smartling-cli` validated configuration at the first step. But later we added more commands with different required parameters and aliases for commands. One more interesting bug happened when you could not run `init` command because you don't have config :).

I would like to reject this request. The order of errors don't matter because you have to fix all of them. Right now flow looks like
```
$ smartling-cli projects
Usage:
  smartling [options] [-v]... init --help
  smartling [options]....

$ smartling-cli projects info
ERROR: no configuration file "smartling.yml" found
```
@enegron-sl what do you think?

**Comment by enegron-sl** [comment]
Agreed.

**Comment by dimitrystd** [comment]
Close it as won't fix.
